.travis.yml

```diff
@@ -2,12 +2,36 @@ language: java
 # Enable container-based infrastructure
 # see http://docs.travis-ci.com/user/workers/container-based-infrastructure/
 sudo: false
-jdk:
-- openjdk6
-- openjdk7
-- oraclejdk8
+addons:
+  apt:
+    packages:
+      - openjdk-6-jdk
+
 cache:
   directories:
   - $HOME/.m2/repository
 script:
-- mvn clean verify
\ No newline at end of file
+- mvn clean verify
+
+matrix:
+  include:
+  - jdk: openjdk6
+    env: CUSTOM_MVN_VERION="3.2.5"
+  - jdk: openjdk7
+    env: CUSTOM_MVN_VERION="3.3.9"
+  - jdk: oraclejdk8
+    env: CUSTOM_MVN_VERION="3.5.0"
+
```

Comment:
Is there really a relationship between Maven version and JDK 8 in particular Maven 3.5.0 has a minimum of JDK 7 to run...?

Reply:
Good question, and I'm sorry I forgot to document my trail.  Short answer: no idea - I copied what worked for another project.

I am not 100% certain, but I'm pretty sure [this](https://github.com/travis-ci/enterprise-installation/issues/14) was the reference I used.  The maven version for JDK6 was the only one I checked - I believe that is the last version supporting jdk6.  The others were simply copied across on the basis of going with something that works, and my lack of familiarity with travis scripting.  It may make sense to use some other version.

In any case, this particularly plugin is unlikely to have a strong need for any particular version of maven.

Reply:
Another solution might be to abandon jdk6 entirely, or compile with JDK7 or 8 with jdk6 compatibility.

Reply:
One other thing as background.  At first I tried simply installing openjdk6, but that failed to execute maven due to an UnsupportedClassVersionError when running maven, so definitely maven had to be downgraded for JDK6.  Now I think about it you could probably delete the env: for JDK7 and JDK8.  Shall I give that a shot?